lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -0,0 +1,30 @@
+<?php
+/**
+ * WP_Theme_JSON_6_1 class
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Class that encapsulates the processing of structures that adhere to the theme.json spec.
+ *
+ * This class is for internal core usage and is not supposed to be used by extenders (plugins and/or themes).
+ * This is a low-level API that may need to do breaking changes. Please,
+ * use get_global_settings, get_global_styles, and get_global_stylesheet instead.
+ *
+ * @access private
+ */
+class WP_Theme_JSON_6_1 extends WP_Theme_JSON_6_0 {
+
+	const ELEMENTS = array(
+		'link' => 'a',
+		'h1'   => 'h1',
+		'h2'   => 'h2',
+		'h3'   => 'h3',
+		'h4'   => 'h4',
+		'h5'   => 'h5',
+		'h6'   => 'h6',
+		// We have the .wp-block-button class so that this will target older buttons that have been serialized.
+		'button' => '.wp-element-button, .wp-block-button',
```

Comment:
I'm not sure if there's a better way to do this...

Reply:
Will that compose well? I don't know the theme.json pipeline enough, but I'd beware of manipulations down the line like this one:

https://github.com/WordPress/gutenberg/blob/39e2aec78079953c8c596a48acb080c88b7860ea/lib/compat/wordpress-5.9/class-wp-theme-json-5-9.php#L541-L549

Notice the concatenation at the end of the snippet. With my superficial knowledge, this smells like it could produce something like:

```css
.my-block-selector .wp-element-button,
.wp-block-button {
}
```

As a workaround, I wondered about grouping with `:is`:

```php
[ 'button' => ':is( .wp-element-button, .wp-block-button )' ]
```

But that could break other aspects of the logic, like equality checks:

https://github.com/WordPress/gutenberg/blob/39e2aec78079953c8c596a48acb080c88b7860ea/lib/compat/wordpress-5.9/class-wp-theme-json-5-9.php#L544